libtenzir/include/tenzir/as_bytes.hpp

```diff
@@ -13,43 +13,54 @@
 #include <array>
 #include <cstddef>
 #include <span>
-#include <type_traits>
 
 namespace tenzir {
 
-inline std::span<const std::byte>
-as_bytes(const void* data, size_t size) noexcept {
+template <size_t Size>
+auto as_bytes(const void* data) noexcept {
+  return std::span<const std::byte, Size>{
+    reinterpret_cast<const std::byte*>(data), Size};
+}
+
+template <size_t Size>
+auto as_writeable_bytes(void* data) noexcept {
+  return std::span<std::byte, Size>{reinterpret_cast<std::byte*>(data), Size};
+}
```

Comment:
This is a really unnecessary optimization that bloats the binary, just use the other two. They'll be inlined by the compiler anyways if you pass in a compile-time constant for the second argument.

Reply:
I tried undoing this but I realized that's actually not redundant. I `span<T, N>` is very much different from a `span<T, dynamic_extent>`. The other overloads do not supporting a fixed-size span, which I need here.